credoai/artifacts.py

```diff
@@ -474,164 +414,144 @@ class CredoData:
-        self.label_key = label_key
+        self.X = X
+        self.y = y
+        self.sensitive_features = sensitive_features
         self.categorical_features_keys = categorical_features_keys
-        self.unused_features_keys = unused_features_keys
-        self.drop_sensitive_feature = drop_sensitive_feature
-        self.nan_strategy = nan_strategy
-        self.X, self.y, self.sensitive_features = self._process_data(self.data).values()
-        self.target_type = type_of_target(self.y)
-
-    def __post_init__(self):
-        self.metadata = self.metadata or {}
         self._validate_data()
-
-    def get_scrubbed_data(self):
-        """Return scrubbed data
-
-        Implements NaN strategy indicated by nan_strategy before returning
-        X, y and sensitive_features dataframes/series.
-
-        Returns
-        -------
-        pd.DataFrame, pd.pd.Series
-            X, y, sensitive_features
-
-        Raises
-        ------
-        ValueError
-            ValueError raised for nan_strategy cannot be used by SimpleImputer
-        """
-        data = self.data.copy()
-        if self.nan_strategy == "drop":
-            data = data.dropna()
-        elif self.nan_strategy == "ignore":
-            pass
-        elif isinstance(self.nan_strategy, str):
-            try:
-                imputer = impute.SimpleImputer(strategy=self.nan_strategy)
-                imputed = imputer.fit_transform(data)
-                data.iloc[:, :] = imputed
-            except ValueError:
-                raise ValueError(
-                    "CredoData's nan_strategy could not be successfully passed to SimpleImputer as a 'strategy' argument"
-                )
-        else:
-            imputed = self.nan_strategy.fit_transform(data)
-            data.iloc[:, :] = imputed
-        return self._process_data(data)
-
-    def _process_data(self, data):
-        # set up sensitive features, y and X
-        y = data[self.label_key]
-        to_drop = [self.label_key]
-        if self.unused_features_keys:
-            to_drop += self.unused_features_keys
-
-        sensitive_features = None
-        if self.sensitive_feature_keys:
-            sensitive_features = data[self.sensitive_feature_keys]
-            if self.drop_sensitive_feature:
-                to_drop.extend(self.sensitive_feature_keys)
-
-        # drop columns from X
-        X = data.drop(columns=to_drop, axis=1)
-        return {"X": X, "y": y, "sensitive_features": sensitive_features}
+        self._process_y()
+        self._process_sensitive(sensitive_intersections)
+        self.X_type = self._determine_X_type()
+        self.y_type = self._determine_y_type()
+
+    def get_data(self):
+        data = {"X": self.X, "y": self.y, "sensitive_features": self.sensitive_features}
+        return data
+
+    def get_X_type(self):
+        return self.X_type
+
+    def get_y_type(self):
+        return self.y_type
+
+    def _determine_X_type(self):
```

Comment:
It seems var types are unused, for both x and y:
`_determine_X_type` -> `self.X_type` -> `get_X_type` -> this is not used anywhere that I can see

Also `self.X_type` is unused  since `isinstance` is used consistently across the class

Reply:
Agreed. Getter methods are unneeded. `X_type` is not intended to just be `isinstance` in the future. More like "tabular"